packages/lab/src/carousel/Carousel.tsx

```diff
@@ -0,0 +1,191 @@
+import { Button, makePrefixer } from "@jpmorganchase/uitk-core";
+import {
+  ChangeEventHandler,
+  Children,
+  forwardRef,
+  HTMLAttributes,
+  ReactElement,
+  useEffect,
+  useState,
+} from "react";
+import { RadioButtonGroup } from "../radio-button";
+import { ChevronLeftIcon, ChevronRightIcon } from "@jpmorganchase/uitk-icons";
+import warning from "warning";
+import { useId } from "../utils";
+import cx from "classnames";
+import "./Carousel.css";
+import {
+  LayoutAnimationDirection,
+  LayoutAnimationTransition,
+} from "@jpmorganchase/uitk-core/src/layout/types";
+import { DeckLayout } from "../layout";
+
+const withBaseName = makePrefixer("uitkCarousel");
+export type SlideDirections = "left" | "right";
+
+export interface CarouselProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * The active Index enables the consumers to select the active slide in the carousel.
+   * Optional, default 0.
+   **/
+  activeIndex?: number;
+  /**
+   * The animation when the slides are shown.
+   * Optional. Defaults to `slide`
+   **/
+  animation?: "slide" | "fade";
+  /**
+   * The timeout for each animation.
+   * Optional. Defaults to 800 ms
+   **/
+  animationTimeout?: number;
+  /**
+   * If this props is passed it will set the aria-label with value to the carousel container.
+   * Optional. Defaults to null
+   */
+  carouselDescription?: string;
+  /**
+   * Collection of slides to render
+   * TODO: Component must implement CarouselSlideProps. Mandatory.
+   */
+  children: ReactElement[];
+  /**
+   * This prop will enable compact / reduced width mode.
+   * The navigation buttons would be part of indicators
+   * Optional. Defaults to false
+   **/
+  compact?: boolean;
+  direction?: LayoutAnimationDirection;
+  /**
+   * It sets the id for the Carousel Container.
+   * String. Optional
+   */
+  id?: string;
+}
+
+const useSlideSelection = (
```

Comment:
Will it be useful for consumers to use?

Reply:
true, even if not for consumers, it might be useful for anything with a carousel, like calendar